### PR TITLE
[bug/feat] Support parameters_to_ignore in DDP

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR resolves a bug with DDP where `parameters_to_ignore` wasn't properly set from module to DDP wrapper.

Fixes #<issue_number>

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

File: tests/accelerators/test_ddp.py [hunk 1]
```diff
@@ -117,3 +120,35 @@ def setup(self, stage: Optional[str] = None) -> None:
...
+
+    class CustomCallback(Callback):
+
+        def on_train_start(self, trainer: 'pl.Trainer', pl_module: 'pl.LightningModule') -> None:
+            assert isinstance(trainer.training_type_plugin.model, DistributedDataParallel)
+            assert trainer.training_type_plugin.model.parameters_to_ignore == ('something')
+            assert trainer.training_type_plugin.model.module._ddp_params_and_buffers_to_ignore == ('something')
+
+    model = CustomModel()
+    trainer = Trainer(fast_dev_run=True, accelerator="ddp", gpus=2, callbacks=CustomCallback())
```
Comment (kaushikb11) [inline tests/accelerators/test_ddp.py:153]: Add `tmpdir` here

File: tests/accelerators/test_ddp.py [hunk 2]
```diff
@@ -117,3 +120,35 @@ def setup(self, stage: Optional[str] = None) -> None:
...
+    class WierdModule(torch.nn.Module):
+
+        def _save_to_state_dict(self, destination, prefix, keep_vars):
+            return {"something": "something"}
+
+    class CustomModel(BoringModel):
+
+        def __init__(self):
+            super().__init__()
+            self.wierd_module = WierdModule()
```
Comment (kaushikb11) [inline tests/accelerators/test_ddp.py:140]:
```suggestion
            self.weird_module = WeirdModule()
```
Idk if it's supposed to Weird or Wired 😂

**Review by ananthsub — APPROVED** [review]
good catch!

File: pytorch_lightning/overrides/base.py [hunk 3]
```diff
@@ -36,6 +36,11 @@ def __init__(self, pl_module: LightningModule):
         super().__init__()
         self.module = pl_module
 
+        # set the parameters_to_ignore from LightningModule.
+        _ddp_params_and_buffers_to_ignore = getattr(pl_module, "_ddp_params_and_buffers_to_ignore", None)
+        if _ddp_params_and_buffers_to_ignore:
+            self._ddp_params_and_buffers_to_ignore = _ddp_params_and_buffers_to_ignore
```
Comment (kaushikb11) [inline pytorch_lightning/overrides/base.py:42]:
```suggestion
        self._ddp_params_and_buffers_to_ignore = getattr(pl_module, "_ddp_params_and_buffers_to_ignore", None)
```
None works as well?

File: tests/accelerators/test_ddp.py (hunk 1 above)
Comment (tchaton) [reply tests/accelerators/test_ddp.py:153]: Done !

**Review by kaushikb11 — APPROVED** [review]
LGTM!

File: tests/accelerators/test_ddp.py (hunk 2 above)
Comment (tchaton) [reply tests/accelerators/test_ddp.py:140]: Done !

File: pytorch_lightning/overrides/base.py (hunk 3 above)
Comment (tchaton) [reply pytorch_lightning/overrides/base.py:42]: Done !

**Review by Borda — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]
